src/server/components/auth/express-session.ts

```diff
@@ -21,35 +21,39 @@ declare module "express-session" {
 }
 
 export async function configureExpressSession(server: Express): Promise<void> {
-  const secret = await getSecretValue(SECRET_NAME);
-
-  setTimeout(() => {
-    rotateSecret(SECRET_NAME).catch(noop);
-  }, random(200) * ONE_MINUTE);
-
-  const options: session.SessionOptions = {
-    secret: secret,
-    saveUninitialized: true,
-    resave: false,
-    proxy: !isLocalHost,
-    cookie: {
-      secure: !isLocalHost,
-      maxAge: isLocalHost ? ONE_DAY : FOUR_HOURS,
-    },
-    name: "lists_sid",
-  };
-
-  if (isRedisAvailable()) {
-    const redisClient = getRedisClient();
-    const RedisStore = connectRedis(session);
-
-    options.store = new RedisStore({
-      client: redisClient,
-      prefix: "lists_session_",
-    });
-
-    logger.info("Redis session storage initialized successfully");
+  try {
+    const secret = await getSecretValue(SECRET_NAME);
+
+    setTimeout(() => {
+      rotateSecret(SECRET_NAME).catch(noop);
+    }, random(200) * ONE_MINUTE);
+
+    const options: session.SessionOptions = {
+      secret: secret,
+      saveUninitialized: true,
+      resave: false,
+      proxy: !isLocalHost,
+      cookie: {
+        secure: !isLocalHost,
+        maxAge: isLocalHost ? ONE_DAY : FOUR_HOURS,
+      },
+      name: "lists_sid",
+    };
+
+    if (isRedisAvailable()) {
+      const redisClient = getRedisClient();
+      const RedisStore = connectRedis(session);
+
+      options.store = new RedisStore({
+        client: redisClient,
+        prefix: "lists_session_",
+      });
+
+      logger.info("Redis session storage initialized successfully");
+    }
+
+    server.use(session(options));
+  } catch (error) {
+    logger.error(`configureExpressSession: Error ${(error as Error).message}`);
```

Comment:
So my assumption (and it could be a wrong one), was there is a chance that `getSecretValue()` could fail. So if it does, the error should be logged somewhere.

Reply:
so this is more than just logging that an error occurred.  In this case, the error handling should throw the error as we don't want the session to be initialised if we can't access SecretsManager.  Instead, the error will be thrown and the user shown an error message.